packages/components/src/hooks/use-tree-view-keyboard-interaction.ts

```diff
@@ -0,0 +1,208 @@
+ * for non-standard behavior, see the `KeyboardInteractionConfiguration` configuration options.
+ */
+export const useTreeViewKeyboardInteraction = ({
+    eventsRoot,
+    focusedItemId,
+    isEndNode,
+    getPrevious,
+    getNext,
+    getFirst,
+    getLast,
+    getParent,
+    getFirstChild,
+    isOpen,
+    open,
+    close,
+    focus,
+    select,
+    endNodeExpandSelectsNext = true, // TODO: move to Codux, default should be false
+    selectionFollowsFocus = true, // TODO: move to Codux, default should be false
+}: TreeViewKeyboardInteractionsParams & KeyboardInteractionConfiguration) => {
+    const handleArrowRight = useCallback(
+        (event: KeyboardEvent) => {
+            if (!focusedItemId) return;
+
+            if (isEndNode(focusedItemId) && !endNodeExpandSelectsNext) {
+                return;
+            } else if (isEndNode(focusedItemId) && endNodeExpandSelectsNext) {
+                const next = getNext(focusedItemId);
+
+                if (next) {
+                    focus(next);
+                    selectionFollowsFocus && select(next);
+                }
+            } else if (isOpen(focusedItemId)) {
+                const firstChild = getFirstChild(focusedItemId);
+
+                if (firstChild) {
+                    focus(firstChild);
+                    selectionFollowsFocus && select(firstChild);
+                }
+            } else {
+                open(focusedItemId);
+            }
+
+            event.preventDefault();
+        },
+        [
+            endNodeExpandSelectsNext,
+            focus,
+            focusedItemId,
+            getFirstChild,
+            getNext,
+            isEndNode,
+            isOpen,
+            open,
+            select,
+            selectionFollowsFocus,
+        ]
+    );
+
+    const handleArrowLeft = useCallback(
+        (event: KeyboardEvent) => {
+            if (!focusedItemId) return;
+
+            if (!isOpen(focusedItemId)) {
+                const parent = getParent(focusedItemId);
+
+                if (parent) {
+                    focus(parent);
+                    selectionFollowsFocus && select(parent);
+                }
+            } else {
+                close(focusedItemId);
+            }
+
+            event.preventDefault();
+        },
+        [close, focus, focusedItemId, getParent, isOpen, select, selectionFollowsFocus]
+    );
+
+    const handleArrowUp = useCallback(
+        (event: KeyboardEvent) => {
+            if (!focusedItemId) return;
+
+            const previous = getPrevious(focusedItemId);
+            if (previous) {
+                focus(previous);
+                selectionFollowsFocus && select(previous);
+            }
+
+            event.preventDefault();
+        },
+        [focus, focusedItemId, getPrevious, select, selectionFollowsFocus]
+    );
+
+    const handleArrowDown = useCallback(
```

Comment:
I would prefer not to do it because the whole idea was for the code to reflex the https://www.w3.org/WAI/ARIA/apg/patterns/treeview/ and how they codify the rules. It is not about code looking pretty, it was about making it align with the spec